### PR TITLE
[SYCL] Enable __SYCL_ID_QUERIES_FIT_IN_INT__ macro to default

### PR DESCRIPTION
**Pull request by smanna12**
This patch enables the macro to default.
The option did not set correctly and  was lost during pulldown
conflict here: https://github.com/intel/llvm/pull/3094/commits/a52d3d3aa6c9ceb3340f1692deec9c0244f73902

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/include/clang/Driver/Options.td [hunk 1]
```diff
@@ -2402,7 +2402,10 @@ def fsycl_device_code_lower_esimd : Flag<["-"], "fsycl-device-code-lower-esimd">
   Flags<[CC1Option, CoreOption]>, HelpText<"Lower ESIMD-specific constructs">;
 def fno_sycl_device_code_lower_esimd : Flag<["-"], "fno-sycl-device-code-lower-esimd">,
   Flags<[CC1Option, CoreOption]>, HelpText<"Do not lower ESIMD-specific constructs">;
-defm sycl_id_queries_fit_in_int: OptInFFlag<"sycl-id-queries-fit-in-int", "Assume", "Do not assume", " that SYCL ID queries fit within MAX_INT.", [CC1Option,CoreOption], LangOpts<"SYCLValueFitInMaxInt">>;
+defm sycl_id_queries_fit_in_int: BoolFOption<"sycl-id-queries-fit-in-int",
+  LangOpts<"SYCLValueFitInMaxInt">, DefaultTrue,
+  PosFlag<SetTrue, [CC1Option], "Assume">, NegFlag<SetFalse, [CC1Option], "Do not assume">,
```
Comment (mdtoguchi) [inline clang/include/clang/Driver/Options.td:2407]:
```suggestion
  PosFlag<SetTrue, [], "Assume">, NegFlag<SetFalse, [], "Do not assume">,
```
`CC1Option` here is redundant as it is already set with `BothFlags`

Comment (smanna12) [reply clang/include/clang/Driver/Options.td]: Thanks @mdtoguchi for the review. I have removed redundant "CC1Option".

**Review by mdtoguchi — APPROVED** [review]

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Review by premanandrao — APPROVED** [review]
LGTM

**Comment by smanna12** [comment]
Thanks for the reviews everyone.

**Comment by smanna12** [comment]
The failures of Jenkins/Precommit  is not related to this patch.

**Comment by bader** [comment]
> The failures of Jenkins/Precommit are not related to this patch.

@kbobrovs, @DenisBakhvalov, could you investigate why ESIMD tests are failing in pre-commit testing, please?

**Comment by DenisBakhvalov** [comment]
> > The failures of Jenkins/Precommit are not related to this patch.
> 
> @kbobrovs, @DenisBakhvalov, could you investigate why ESIMD tests are failing in pre-commit testing, please?

I can't see failures. In CI I can see `script returned exit code 255`. 

Also, I remember there was a problem with enabling `__SYCL_ID_QUERIES_FIT_IN_INT__` for ESIMD because of `llvm.assume` intrinsic generated in the device code. Vector backend can't yet handle those.

**Comment by bader** [comment]
> > > The failures of Jenkins/Precommit are not related to this patch.
> > 
> > 
> > @kbobrovs, @DenisBakhvalov, could you investigate why ESIMD tests are failing in pre-commit testing, please?
> 
> I can't see failures. In CI I can see `script returned exit code 255`.
> 
> Also, I remember there was a problem with enabling `__SYCL_ID_QUERIES_FIT_IN_INT__` for ESIMD because of `llvm.assume` intrinsic generated in the device code. IGC Vector backend can't yet handle those.

Here is the link to the log: http://icl-jenkins.sc.intel.com:8080/blue/rest/organizations/jenkins/pipelines/SYCL_CI/pipelines/intel/pipelines/Lin/pipelines/LLVM_Test_Suite/runs/2217/nodes/27/steps/335/log/?start=0

It might be caused by https://github.com/intel/llvm/pull/3242, which was reverted. I'll re-run the test to validate this guess.

**Comment by bader** [comment]
> It might be caused by #3242, which was reverted. I'll re-run the test to validate this guess.

@tfzhu, @vladimirlaz, it looks like simple re-run doesn't help. Do you know if PR branch is merged with the tip of the `sycl` branch before running `Jenkins/Precommit`? If no, how can we test this PR with up-to-date `sycl` branch?

**Comment by DoyleLi** [comment]
> > It might be caused by #3242, which was reverted. I'll re-run the test to validate this guess.
> 
> @tfzhu, @vladimirlaz, it looks like simple re-run doesn't help. Do you know if PR branch is merged with the tip of the `sycl` branch before running `Jenkins/Precommit`? If no, how can we test this PR with up-to-date `sycl` branch?

Hi @bader, Jenkins/Precommit is merged with top of sycl branch. Here take latest result for example:
http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLin%2FLLVM_Test_Suite/detail/LLVM_Test_Suite/2258/pipeline/116

See complete log at:
http://icl-jenkins.sc.intel.com:8080/blue/rest/organizations/jenkins/pipelines/SYCL_CI/pipelines/intel/pipelines/Lin/pipelines/LLVM_Test_Suite/runs/2258/nodes/116/steps/121/log/?start=0

Currently it merged with 2a7a1496e3062fd585f88d86ade190096a990659 in sycl branch.

**Comment by bader** [comment]
@DoyleLi, thanks!

@smanna12, it seems that this patch introduces regressions to ESIMD tests. Please, take a look.

**Comment by smanna12** [comment]
> @DoyleLi, thanks!
> 
> @smanna12, it seems that this patch introduces regressions to ESIMD tests. Please, take a look.

Thanks @bader, i will take a look at this.

**Comment by bader** [comment]
> > @DoyleLi, thanks!
> > @smanna12, it seems that this patch introduces regressions to ESIMD tests. Please, take a look.
> 
> Thanks @bader, i will take a look at this.

@smanna12, do you have any updates?

**Pull request closed without merging**